### DISCUSSION TITLE
Conversation memory control

### DISCUSSION THREAD

**Discussion by shenghann** [ajndkr/lanarky#68] [Q&A]
I have set up an agent executor initialized with ConversationBufferMemory that serves a websocket endpoint. However, not very sure about how should we manage and compartmentalize memory with a unique key for each user in this pattern. Any suggestions?

**Comment by ajndkr** [comment]
hmm. I don't have much experience with websocket solutions. One way to manage memory is to update the frontend code to send all messages in a conversation which can be used to update the `memory` buffer on a per-request basis.

Otherwise, you'll require a database + frontend code changes to track conversations via a session id or something similar.
